Port Font enums to the new IPC serialization format

csaavedra opened the pull request:
#### d3671f566b6e5bfc9a31a795d336b43d17c1c1e3
<pre>
Port Font enums to the new IPC serialization format
<a href="https://bugs.webkit.org/show_bug.cgi?id=264883">https://bugs.webkit.org/show_bug.cgi?id=264883</a>

Reviewed by Chris Dumez.

Move the serializable enums out of the Font class, remove
the EnumTraits bits, and port the serialization to the
generator format.

Also take the opportunity to rename Font::Interstitial
and Font::OrientationFallback to Font::IsInterstitial
and Font::IsOrientationFallback respectively, to better
match their boolean nature.

* Source/WebCore/css/CSSFontFace.cpp:
(WebCore::CSSFontFace::font):
* Source/WebCore/platform/graphics/Font.cpp:
(WebCore::Font::create):
(WebCore::Font::Font):
(WebCore::m_shouldNotBeUsedForArabic):
(WebCore::Font::verticalRightOrientationFont const):
(WebCore::Font::uprightOrientationFont const):
(WebCore::Font::invisibleFont const):
(WebCore::Font::brokenIdeographFont const):
* Source/WebCore/platform/graphics/Font.h:
(WebCore::Font::isTextOrientationFallback const):
(WebCore::Font::isInterstitial const):
* Source/WebCore/platform/graphics/freetype/SimpleFontDataFreeType.cpp:
(WebCore::Font::platformCreateScaledFont const):
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
(IPC::ArgumentCoder&lt;Font::Attributes&gt;::decode):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: <a href="https://commits.webkit.org/270827@main">https://commits.webkit.org/270827@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f88bf0b42738bcc976fe65c10384cb1c16ae5c8d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/26500 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/5115 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/27747 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/28690 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/24236 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/26907 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/6916 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/2523 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/24195 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/26760 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/3939 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/22758 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/3470 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/3514 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/23751 "Found 1 new test failure: platform/mac/media/encrypted-media/fps-generateRequest.html (failure)") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/29194 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/24153 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/24138 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/29785 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/3551 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/1749 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/27679 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/4985 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6367 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/4026 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/3876 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/platform/graphics/Font.h:
```diff
@@ -78,11 +84,11 @@ DECLARE_ALLOCATOR_WITH_HEAP_IDENTIFIER(Font);
-    enum class Visibility : bool { Visible, Invisible };
-    enum class OrientationFallback : bool { No, Yes };
+    using Origin = FontOrigin;
+    using Interstitial = FontInterstitial;
```
Not your fault but since we're changing this. The name isn't good. This should be named `IsInterstitial / FontIsInterstitial`.

csaavedra replied:
I agree, I actually thought about that. I'll change it.

cdumez commented on Source/WebCore/platform/graphics/Font.h:
```diff
@@ -78,11 +84,11 @@ DECLARE_ALLOCATOR_WITH_HEAP_IDENTIFIER(Font);
+    using Origin = FontOrigin;
+    using Interstitial = FontInterstitial;
+    using Visibility = FontVisibility;
+    using OrientationFallback = FontOrientationFallback;
```
Not your fault but since we're changing this. The name isn't good. This should be named `IsOrientationFallback / FontIsOrientationFallback`.

csaavedra replied:
Same!

csaavedra committed:
Port Font enums to the new IPC serialization format
https://bugs.webkit.org/show_bug.cgi?id=264883

Reviewed by Chris Dumez.

Move the serializable enums out of the Font class, remove
the EnumTraits bits, and port the serialization to the
generator format.

Also take the opportunity to rename Font::Interstitial
and Font::OrientationFallback to Font::IsInterstitial
and Font::IsOrientationFallback respectively, to better
match their boolean nature.

* Source/WebCore/css/CSSFontFace.cpp:
(WebCore::CSSFontFace::font):
* Source/WebCore/platform/graphics/Font.cpp:
(WebCore::Font::create):
(WebCore::Font::Font):
(WebCore::m_shouldNotBeUsedForArabic):
(WebCore::Font::verticalRightOrientationFont const):
(WebCore::Font::uprightOrientationFont const):
(WebCore::Font::invisibleFont const):
(WebCore::Font::brokenIdeographFont const):
* Source/WebCore/platform/graphics/Font.h:
(WebCore::Font::isTextOrientationFallback const):
(WebCore::Font::isInterstitial const):
* Source/WebCore/platform/graphics/freetype/SimpleFontDataFreeType.cpp:
(WebCore::Font::platformCreateScaledFont const):
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
(IPC::ArgumentCoder<Font::Attributes>::decode):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: https://commits.webkit.org/270827@main
